tools/mo/openvino/tools/mo/front/extractor.py

```diff
@@ -1051,8 +1053,20 @@ def add_input_ops(graph: Graph, user_defined_inputs: dict, before_infer: bool):
                 user_shape = None
                 if shape is not None:
                     user_shape = shape
-                    shape = shape_array(
-                        [dim if type(dim) != tuple and dim >= 0 else dynamic_dimension_value for dim in shape])
+                    shape_list = []
+                    for dim in shape:
+                        if isinstance(dim, Dimension):
+                            if dim.is_static:
+                                shape_list.append(dim.get_min_length())
+                            else:
+                                shape_list.append(dynamic_dimension_value)
+                            continue
+                        if dim >= 0:
+                            shape_list.append(dim)
+                        else:
+                            shape_list.append(dynamic_dimension_value)
+
```

Comment:
Can we provide a dictionary of `PartialShape` to this function? so we will not have the logic for `list`. We can normalize arguments to this function preliminary

Reply:
Here only one shape of type `PartialShape `is processed, it can be treated as list of `Dimension's`. Then list of `int `values is created (`dynamic_dimension_value `for dynamic dims) and passed to `shape_array`() to create masked array. 
So dictionary of `PartialShape `is not applicable here.